src/presentation/service/runtime.go

```diff
@@ -0,0 +1,183 @@
+	}
+
+	return NewServiceOutput(Success, phpConfigs)
+}
+
+func (service *RuntimeService) parsePhpModules(
+	rawPhpModules interface{},
+) ([]entity.PhpModule, error) {
+	modules := []entity.PhpModule{}
+
+	rawModulesSlice, assertOk := rawPhpModules.([]interface{})
+	if !assertOk {
+		return modules, errors.New("InvalidPhpModules")
+	}
+
+	for _, rawModule := range rawModulesSlice {
+		rawModuleMap, assertOk := rawModule.(map[string]interface{})
+		if !assertOk {
+			slog.Debug("PhpModuleIsNotAnInterface")
+			continue
+		}
+
+		moduleName, err := valueObject.NewPhpModuleName(rawModuleMap["name"])
+		if err != nil {
+			slog.Debug("InvalidPhpModuleName", slog.Any("name", rawModuleMap["name"]))
+			continue
+		}
+
+		moduleStatus, assertOk := rawModuleMap["status"].(bool)
+		if !assertOk {
+			slog.Debug(
+				"InvalidPhpModuleStatus", slog.Any("status", rawModuleMap["status"]),
+			)
+			continue
+		}
+
+		modules = append(modules, entity.NewPhpModule(moduleName, moduleStatus))
+	}
+
+	return modules, nil
+}
+
+func (service *RuntimeService) parsePhpSettings(
+	rawPhpSettings interface{},
+) ([]entity.PhpSetting, error) {
+	settings := []entity.PhpSetting{}
+
+	rawSettingsSlice, assertOk := rawPhpSettings.([]interface{})
+	if !assertOk {
+		return settings, errors.New("InvalidPhpSettings")
+	}
+
+	for _, rawSetting := range rawSettingsSlice {
+		rawSettingMap, assertOk := rawSetting.(map[string]interface{})
+		if !assertOk {
+			slog.Debug("PhpSettingIsNotAnInterface")
+			continue
+		}
+
+		settingName, err := valueObject.NewPhpSettingName(rawSettingMap["name"])
+		if err != nil {
+			slog.Debug(
+				"InvalidPhpSettingName", slog.Any("name", rawSettingMap["name"]),
+			)
+			continue
+		}
+
+		settingValue, err := valueObject.NewPhpSettingValue(rawSettingMap["value"])
+		if err != nil {
+			slog.Debug(
+				"InvalidPhpSettingValue", slog.Any("value", rawSettingMap["value"]),
+			)
+			continue
+		}
+
+		settings = append(
+			settings,
+			entity.NewPhpSetting(
+				settingName, settingValue, []valueObject.PhpSettingOption{},
+			),
+		)
+	}
+
+	return settings, nil
+}
+
+func (service *RuntimeService) UpdatePhpConfigs(
+	input map[string]interface{},
+) ServiceOutput {
+	serviceName, _ := valueObject.NewServiceName("php-webserver")
+	sharedHelper.StopIfServiceUnavailable(service.persistentDbSvc, serviceName)
+
+	requiredParams := []string{"hostname", "version"}
+	err := serviceHelper.RequiredParamsInspector(input, requiredParams)
+	if err != nil {
+		return NewServiceOutput(UserError, err.Error())
+	}
+
+	hostname, err := valueObject.NewFqdn(input["hostname"])
+	if err != nil {
+		return NewServiceOutput(UserError, err.Error())
+	}
+
+	phpVersion, err := valueObject.NewPhpVersion(input["version"])
+	if err != nil {
+		return NewServiceOutput(UserError, err.Error())
+	}
+
+	phpModules := []entity.PhpModule{}
+	if input["modules"] != nil {
+		phpModules, err = service.parsePhpModules(input["modules"])
```

Comment:
How would someone know input["modules"] is a []interface{}? They wouldn't cause service is used by multiple types of controllers. That's implicit knowledge that you won't remember in a few months.

Reply:
Fixed by changing the request change above.